controllers/workspace/devworkspace_controller.go

```diff
@@ -361,11 +334,20 @@ func (r *DevWorkspaceReconciler) stopWorkspace(workspace *devworkspace.DevWorksp
 
 	if workspaceDeployment.Status.Replicas == 0 {
 		logger.Info("Workspace stopped")
-		status.Phase = devworkspace.WorkspaceStatusStopped
+		status.phase = devworkspace.WorkspaceStatusStopped
 	}
 	return r.updateWorkspaceStatus(workspace, logger, status, reconcile.Result{}, nil)
 }
 
+// failWorkspace marks a workspace as failed by setting relevant fields in the status struct. These changes are not
+// propagated to the cluster, and must be synced via updateWorkspaceStatus()
```

Comment:
if it's not propagated to the cluster, what is the reason to return `reconcile.Result` here?

Reply:
So that you can call `return failWorkspace()` instead of `return reconcile.Result{}, failWorkspace()` in the reconcile loop.

Reply:
OK, then changes must be synced with updateWorkspaceStatus only in when client needs these changes by flushed before `return reconcile.Result{}`

Then maybe doc should be
```
// These changes are not synced to cluster immediately.
// The cluster will be reflected with the made changes after reconcile loop finished.
// Or if needed, it can be flushed via updateWorkspaceStatus()
```
WDYT?

Reply:
Makes sense, will do.